lib/src/kvs/speedb/mod.rs

```diff
@@ -136,6 +137,60 @@ impl Transaction {
 		// Return result
 		Ok(res)
 	}
+	/// Obtain a new change timestamp for a key
+	/// which is replaced with the current timestamp when the transaction is committed.
+	/// NOTE: This should be called when composing the change feed entries for this transaction,
+	/// which should be done immediately before the transaction commit.
+	/// That is to keep other transactions commit delay(pessimistic) or conflict(optimistic) as less as possible.
+	#[allow(unused)]
+	pub async fn get_timestamp<K>(&mut self, key: K) -> Result<Versionstamp, Error>
+	where
+		K: Into<Key>,
+	{
+		// Check to see if transaction is closed
+		if self.ok {
+			return Err(Error::TxFinished);
+		}
+		// Write the timestamp to the "last-write-timestamp" key
+		// to ensure that no other transactions can commit with older timestamps.
+		let k: Key = key.into();
+		let prev = self.tx.lock().await.as_ref().unwrap().get_opt(k.clone(), &self.ro)?;
+		let ver = match prev {
+			Some(prev) => {
+				let slice = prev.as_slice();
+				let res: Result<[u8; 10], Error> = match slice.try_into() {
+					Ok(ba) => Ok(ba),
+					Err(e) => Err(Error::Ds(e.to_string())),
+				};
+				let array = res?;
+				let prev = to_u64_be(array);
+				prev + 1
+			}
+			None => 1,
+		};
+
+		let verbytes = u64_to_versionstamp(ver);
+
+		self.tx.lock().await.as_ref().unwrap().put(k, verbytes)?;
```

Comment:
do you need to check for `self.tx` to be writable? if yes, I guess it applies to all the `get_timestamp` methods defined for other datastores

Reply:
Great question! Yeah probably. I honestly don't fully understand the consequence of having these prechecks in the beginning of functions that writes KVs! To make things consistent with existing functions though, adding the writeability check does make sense.

In the latest commit, I fixed `get_versionstamped_key` rather than `get_timestamp`. The former is supposed to be used only in a writeable transaction whereas the latter can be used even in a read-only transaction.